test/e2e/cloud_provider_kubevirt/load_balancer_test.go

```diff
@@ -0,0 +1,139 @@
+package cloud_provider_kubevirt
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"kubevirt.io/cloud-provider-kubevirt/test/resources"
+)
+
+const namespace = "default"
+const testAppName = "test-app"
+
+var _ = Describe("Load Balancer", func() {
+	var (
+		err                   error
+		server                *appsv1.Deployment
+		service               *v1.Service
+		curlJob               *batchv1.Job
+		backgroundPropagation = metav1.DeletePropagationBackground
+	)
+
+	BeforeEach(func() {
+		server = resources.HTTPServerDeployment(testAppName, namespace)
+		service = resources.HTTPServerService(testAppName, namespace)
+	})
+	Context("when a LB service is created in tenant cluster", func() {
+		BeforeEach(func() {
+			err = tenantClient.Create(context.TODO(), server)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() bool {
+				err = tenantClient.Get(context.TODO(), namespacedName(server), server)
+				Expect(err).NotTo(HaveOccurred())
+				return server.Status.ReadyReplicas == *server.Spec.Replicas
+			}, 60*time.Second, time.Second).Should(BeTrue())
+
+			err = tenantClient.Create(context.TODO(), service)
+			Expect(err).NotTo(HaveOccurred())
+
+			DeferCleanup(func() {
+				err = tenantClient.Delete(context.TODO(), server)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(func() error {
+					return tenantClient.Get(context.TODO(), namespacedName(server), server)
+				}, 120*time.Second, time.Second).Should(WithTransform(errors.IsNotFound, BeTrue()))
```

Comment:
Can we do something like this to generalize the delete function ? (although I rememer it may have problems detecting the type at .Delete)

```golang
func deleteResource(cli client.Client, namespace string,  resouce obj client.Object) {
   err = cli.Delete(context.TODO(), resource)
   Expect(err).NotTo(HaveOccurred())
   Eventually(func() error {
      return tenantClient.Get(context.TODO(), namespacedName(resouce), resouce)
   }, 120*time.Second, time.Second).Should(WithTransform(errors.IsNotFound, BeTrue()))
}
```

```golang
deleteResource(tenantClient, server)
deleteResource(tenatnClient, service)
```

Reply:
Thanks, I think it's a good idea, see [commit](https://github.com/kubevirt/cloud-provider-kubevirt/pull/109/commits/f30c1d886eeb9025476a68a588a6adf37b5ac6f3)